hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HealthTransformations.scala

```diff
@@ -68,8 +67,42 @@ object HealthTransformations {
       None
     }
 
+  def mapEyeDisease(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
+    if (rawRecord.getBoolean("hs_dx_eye_yn")) {
+      eye_diseases.flatMap {
+        // "other" case
+        case ("eye_other", categorical) =>
+          createHealthConditionRow(
+            rawRecord,
+            "dx_eye_other",
+            eyeDiseaseCondition,
+            categorical,
+            conditionOtherDescription = rawRecord.getOptional("hs_dx_eye_other_spec")
+          )
+        case ("blind", categorical) => {
+          val isCauseKnown = rawRecord.getBoolean("hs_dx_eye_cause_yn")
+          val conditionCause = if (isCauseKnown) { rawRecord.getOptionalNumber("hs_dx_eye_cause") }
+          else { None }
+          createHealthConditionRow(
+            rawRecord,
+            "dx_blind",
+            eyeDiseaseCondition,
+            categorical,
+            conditionCause = conditionCause,
+            conditionCauseOtherDescription =
+              if (isCauseKnown && (conditionCause.getOrElse(None) == 98)) {
+                rawRecord.getOptional("hs_dx_eye_cause_other")
+              } else { None }
+          )
+        }
+        case (disease, categorical) =>
+          createHealthConditionRow(rawRecord, s"dx_${disease}", eyeDiseaseCondition, categorical)
+      }
+    } else { None }
+
   // list conditions and assign categorical Longs
   val infectiousDiseaseCondition = 1L
+  val eyeDiseaseCondition = 2L
```

Comment:
According to the RedCap layout, I think eyes should have value 1 (they come first in the ordering of questions)

Reply:
Where is the layout? I was going by order of stuff [here](https://redcap.fredhutch.org/redcap_v9.8.2/Design/data_dictionary_codebook.php?pid=895), where hs_dx_infectious_yn is at number 1275 but hs_dx_eye_yn is at 1623

Reply:
The congenital things are asked first, so I've been thinking we'd do a partial-overlay style of ordering: Assign indexes based on the CG ordering, then add the diagnosed-only conditions to the end.

Reply:
Makes sense; I am still coming down from the shock of learning that there are completely differently named congenital eye things =P